.github/workflows/test_push.yaml

```diff
@@ -0,0 +1,37 @@
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+env:
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Why is this needed?

Reply:
For the coveralls plugin, to be able to authenticate there.
This is an automatic token generated by github: https://docs.github.com/en/actions/security-guides/automatic-token-authentication